controls.py

```diff
@@ -5,64 +5,57 @@
 
 
 ######################################Creation Functions
-def create_student(username, name, password, email, token=""):
+def create_account(username):
     # Create a Student and Account instance in the rd database.
     s = db.get_session()
 
-    Acc =  Account(username=username, name=name, password=password, email=email, account_token=token)
+    Acc =  Account(username)
     s.add(Acc)
-    s.flush()
-
-    Stu = Student(related_account=Acc.id)
-    s.add(Stu)
     s.commit()
     s.close()
+
     print(username, "added to the database")
 
-def create_donor(username, password, email, token=""):
-    # Create a Donor and Account instance in the rd database
+def approve_admin(usernameAttempt):
     s = db.get_session()
-
-    Acc =  Account(username=username, password=password, email=email, account_token=token)
-    s.add(Acc)
-    s.flush()
-
-    Don = Donor(related_account=Acc.id)
-    s.add(Don)
+    A = s.query('accounts').filter(Account.username==usernameAttempt).first()
+    A.admin = True
     s.commit()
     s.close()
 
+def approve_requesting(usernameAttempt):
+    s = db.get_session()
+    A = s.query('accounts').filter(Account.username==usernameAttempt).first()
+    A.approved = True
+    s.commit()
+    s.close()
 
-def open_request(student_id, amount, reason, title="Donate to a Reedie in needie", anon=True, amount_filled=0.0):
+def open_request(student_id, amount, reason, title, anon):
     # Create a request instance in the rd database.
     s = db.get_session()
-    req = Request(requested_by=student_id, amount_needed=amount, title=title, amount_filled=amount_filled, description=reason, anon=anon)
+    if title == '': title = "Donate to a Reedie in needie"
+    req = Request(student_id, amount, title, reason, anon, True)
     s.add(req)
     s.commit()
     s.close()
     print(student_id, "request added to the database")
-       
 
-def donate(request_id, donor_id, amount, account_token):
+def donate(username, request_id, amount):
     # Create and log the effects of a Donation in the database
     s = db.get_session()
-    D = s.query(Donor).get(donor_id)
-    A = s.query(Account).get(D.related_account)
-    R = s.query(Request).get(request_id)
-
+    A = s.query(Account).filter(Account.username==usernameAttempt).first()
+    R = s.query(Request).filter(request.id==request_id).first()
 
-    A.account_token = account_token
-    D.money_given += amount
-    D.donations_given += 1
+    A.given += amount
+    A.num_given += 1
+    s.flush()
     R.amount_filled += amount
 
     # check to see if the request has been completed
     if R.amount_needed >= R.amount_filled:
         R.filled = True

```

Comment:
is this where we should start referencing stripe data? or do we even need to do that really?